### ISSUE TITLE
[HELP]: Filter configuration not working

### ISSUE THREAD

**Issue by wimille** [qrti/funcList#15]
Hi,

I recently discovered your extension, and i started to test it. I develop FPGA by using VHDL language.
So i try to setup my own filter, but it doesn't work.

Here's a very simple line from my file:
```
signal toto : std_logic;
```

And my filter:
```
"funcList": {
    "doubleSpacing": false,

    "filters": [
      {
        "extensions": [
          ".vhd",
          ".vhdl"
        ],
        "native": "/^\\s*signal\\s+\\w+/mgi",
        "display": "/.*/"
        "sort": 0
      }
    ]
  }
```

I try my regex with an online website, and it works fine. But when i call the "show function", a new tab appears with no matches.

I don't see my mistakes.

Best regards,
Mathieu

**Comment by qrti** [comment]
I tried your example, in the new tab 'signal toto' is shown, it seems to work for me

thoughts
- it is necessary to save the source file as funcList registered it, in your case .vhd or .vhdl
- your filter misses a comma before "sort"

**Comment by wimille** [comment]
Hi

First, you've right, i missed a comma in my filter configuration.

Yesterday, for some reason i had to log off (and so quit VSCode).
Then, i've reopened my VHDL File and launched "show function", and it worked.

Maybe i miss something, but i have to close VSCode for the extenstion take my filters updates into account. It is not very easy to customatize filters with this method.

**Comment by qrti** [comment]
hm, as far as I can tell filters are updated as soon as saving the .json file, I'm working on a windows 10 system

**Comment by wimille** [comment]
Hi, it's weird because i save my json file after each update. For now, although it is not very easy i can customize my filters.
I work on Windows 7.

**Comment by qrti** [comment]
you are right, I tried again, you have to restart vscode to update a changed filter
an other way is to close and reopen the filtered file

**Comment by wimille** [comment]
I have some others questions:
1. Is it possible to have many filters by file extension ?
For example, in my case, i'd like to list functions and signals for my VHDL files. To do this, i'd need two filters.

2. It is possible to add some extra-comment on display?
For example, if i filter signals and functions, is it possible to "split" those types ?

**Comment by qrti** [comment]
both points not possible, but good thoughts, I put it on my list ;) ...

**Comment by waikin714** [comment]
thanks for both, I have encounter the same problem, which is the setting doesn't work. After read this post, I reopen the vscode, and it works now!
